Userland/Libraries/LibWeb/HTML/HTMLScriptElement.cpp

```diff
@@ -558,6 +571,153 @@ void HTMLScriptElement::inserted()
     HTMLElement::inserted();
 }
 
+// https://html.spec.whatwg.org/multipage/webappapis.html#parse-an-import-map-string
+ErrorOr<ImportMap> HTMLScriptElement::parse_an_import_map_string()
+{
+    // 1. Let parsed by the result of parsing a JSON string to an Infra value given input.
+    auto parsed = TRY(JsonValue::from_string(child_text_content().view()));
+    // 2. FIXME If parsed is not an ordered map, then throw a TypeError indicating that the top-level value needs to be a JSON object.
+    if (!parsed.is_object()) {
```

Comment:
I'm not 100% certain but I think a JsonObject fits the definition of an ordered map. We also want to consider `JsonValue::from_string()` failing as "not an ordered map", so this would become:

```suggestion
    // 1. Let parsed by the result of parsing a JSON string to an Infra value given input.
    auto parsed = JsonValue::from_string(child_text_content().view());
    // 2. If parsed is not an ordered map, then throw a TypeError indicating that the top-level value needs to be a JSON object.
    if (parsed.is_error() || !parsed.is_object()) {
```